party_time/party_time.lua

```diff
@@ -0,0 +1,404 @@
+        ['remove'] = 'difference',
+        ['rm'] = 'difference',
+        ['r'] = 'difference',
+        ['-'] = 'difference',
+    },
+    option = {
+        ['ask'] = 'ask',
+        ['a'] = 'ask',
+        ['whitelist'] = 'whitelist',
+        ['w'] = 'whitelist',
+        ['blacklist'] = 'blacklist',
+        ['b'] = 'blacklist',
+    },
+    bool = {
+        ['1'] = true,
+        ['true'] = true,
+        ['t'] = true,
+        ['yes'] = true,
+        ['y'] = true,
+        ['on'] = true,
+        ['0'] = false,
+        ['false'] = false,
+        ['f'] = false,
+        ['no'] = false,
+        ['n'] = false,
+        ['off'] = false,
+    },
+}
+
+command.arg.register('name', '<name:string(%a+)>')
+
+command.arg.register_type('lookup_boolean', {
+    check = function(str)
+        local bool = arg_lookups.bool[str]
+        if bool ~= nil then
+            return bool
+        end
+
+        error('Expected one of \'' .. table.concat(args_lookups.bool, ',') .. '\', received \'' .. str .. '\'.')
+    end
+})
+
+command.arg.register_type('lookup_add_remove', {
+    check = function(str)
+        local op = arg_lookups.addremove[str]
+        if op ~= nil then
+            return op
+        end
+
+        error('Expected one of \'' .. table.concat(args_lookups.addremove, ',') .. '\', received \'' .. str .. '\'.')
+    end
+})
+command.arg.register_type('lookup_option', {
+    check = function(str)
+        local option = arg_lookups.option[str]
+        if option ~= nil then
+            return option
+        end
+
+        error('Expected one of \'' .. table.concat(args_lookups.option, ',') .. '\', received \'' .. str .. '\'.')
+    end
+})
+
+-- Addon Command Handlers
+-- Seven main commands additoinal sub_commands within
+local pt = command.new('pt')
+
+local invite = function(...)
+    for _, name in pairs({...}) do
+        command.input('/pcmd add ' .. name)
+        name = name:gsub('^%l', string.upper)
+        if unhandled_requests[name] then
+            unhandled_requests[name] = nil
+        end
+    end
+end
+
+pt:register('i', invite, '{name}*')
+pt:register('invite', invite, '{name}*')
+
+
+local request = function(name)
+    command.input('/prcmd add ' .. name)
+end
+
+pt:register('r', request, '{name}')
+pt:register('request', request, '{name}')
+
+
+local blacklist = function(sub_cmd, ...)
+    local names = sets({...})
+    options.blacklist[sub_cmd](options.blacklist, names)
+    settings.save(options)
+end
+
+pt:register('b', blacklist, '<sub_cmd:lookup_add_remove> {name}*')
+pt:register('blacklist', blacklist, '<sub_cmd:lookup_add_remove> {name}*')
+
+
+local whitelist = function(sub_cmd, ...)
+    local names = sets({...})
+    options.whitelist[sub_cmd](options.whitelist, names)
+    settings.save(options)
+end
+
+pt:register('w', whitelist, '<sub_cmd:lookup_add_remove> {name}*')
+pt:register('whitelist', whitelist, '<sub_cmd:lookup_add_remove> {name}*')
+
+
+local ui_enable = function(bool)
```

Comment:
Sorry I didn't comment on this earlier, but I'm wondering if we really need that. Without a UI the user would miss (what I consider) big features, like adding people to the blacklist and whitelist. And I also have plans to expand UI usage (though I didn't comment on that yet, to not delay the PR even further :D). I did mention one thing I'd like to see further down, a popup about the treasure pool preventing auto-joining. The only way this variable would make sense is to disable all of the GUI features, which... I can imagine some people wanting, but I don't know if it's really *good*, even if we let them have their way.

The primary reason I can see to avoid the popup dialogues is because you want to auto-join, and you already achieve that by either being in blacklist mode or having that person on your whitelist.

I'm curious what other people think about this, let's discuss this in Discord.